docs/Reference/Task-Statuses.md

```diff
@@ -0,0 +1,309 @@
+      - `OverrideDependencies != true`
+      - The task has unattainable dependencies (`isUnattainable == true`)
+    - **Display Status**: `blocked`
+
+11. **Will Run**:
+    - **Condition**:
+      - `Status == "undispatched"`
+      - `Activated == true`
+    - **Display Status**: `will-run`
+
+12. **Default**:
+    - **Condition**: If none of the above conditions are met.
+    - **Display Status**: The task's original status (`Status`)
+
+### Display Statuses
+
+Below are the display statuses derived from the logic above, along with their
+descriptions.
+
+---
+
+#### `known-issue`
+
+**Description**: The task has failed and is being tracked by a linked issue in
+the task annotations.
+
+**Usage**: Helps users quickly identify failures related to recognized problems.
+
+---
+
+#### `aborted`
+
+**Description**: The task was aborted while it was running.
+
+**Usage**: Used when a task is intentionally stopped before completion.
+
+---
+
+#### `success`
+
+**Description**: The task has finished successfully without any errors.
+
+**Usage**: Denotes tasks that have met all requirements and passed all tests.
+
+---
+
+#### `setup-failed`
+
+**Description**: The task failed during the setup phase.
+
+**Usage**: Highlights failures occurring before the main execution, useful for
+setup diagnostics.
+
+---
+
+#### `system-unresponsive`
+
+**Description**: The task failed due to a system error, specifically a heartbeat
+timeout indicating the system became unresponsive.
+
+**Usage**: Differentiates failures due to unresponsive systems for targeted
+troubleshooting.
+
+---
+
+#### `system-timed-out`
+
+**Description**: The task failed due to a system-level timeout.
+
+**Usage**: Indicates tasks that failed because the system exceeded the allowed
+time for operations.
+
+---
+
+#### `system-failed`
+
+**Description**: The task failed due to a system error.
+
+**Usage**: Differentiates system-level failures from other failure types for
+troubleshooting.
+
+---
+
+#### `test-timed-out`
+
+**Description**: The task failed because one or more tests exceeded the allotted
+time (`Details.TimedOut == true`).
+
+**Usage**: Indicates tasks that did not complete due to time constraints during
+testing phases.
+
+---
+
+#### `unscheduled`
+
+**Description**: The task is undispatched and not scheduled to run
+(`Activated == false` and `Status == "undispatched"`).
+
+**Usage**: Helps users identify tasks that are inactive and won't be executed.
+
+---
+
+#### `blocked`
+
+**Description**: The task cannot run because it is blocked by unattainable
+dependencies.
+
+**Usage**: Alerts users to tasks that are pending but cannot proceed due to
+unmet dependencies.
+
+---
+
+#### `will-run`
+
+**Description**: The task is scheduled to run eventually, assuming all
+dependencies are met (`Activated == true` and `Status == "undispatched"`).
+
+**Usage**: Indicates tasks in the queue that are pending execution upon
+dependency resolution.
+
+---
+
+### `pending`
```

Comment:
I have never seen this, does this exist?

Reply:
This is the state a task dependency can be in but not the task itself. https://github.com/evergreen-ci/evergreen/blob/4394a9d14c4c485f76e0af36fd2a5d2ed3a88ed6/service/task.go#L452